mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -71,6 +71,15 @@ private[clustering] trait KMeansParams extends Params with HasMaxIter with HasFe
   @Since("1.5.0")
   def getInitMode: String = $(initMode)
 
+  @Since("2.3.0")
+  final val distanceMeasure = new Param[String](this, "distanceMeasure", "The distance measure. " +
```

Comment:
Interesting question here -- what about supplying a function as an argument, for full generality? but then that doesn't translate to Pyspark I guess, and, probably only 2-3 distance functions that ever make sense here.

Reply:
This would be hard for two main reasons:
 1 - as I will explain later, even though theoretically we would need only a function, in practice this is not true for performance reasons;
 2 - saving and loading a module would be much harder (I'm not sure it would even be feasible).